schemas/content/product-attributes.schema.json

```diff
@@ -0,0 +1,101 @@
+{
+  "meta:license": [
+    "Copyright 2018 Adobe Systems Incorporated. All rights reserved.",
+    "This work is licensed under a Creative Commons Attribution 4.0 International (CC BY 4.0) license",
+    "you may not use this file except in compliance with the License. You may obtain a copy",
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$id": "https://ns.adobe.com/xdm/content/product-attributes",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "Product Attributes",
+  "type": "object",
+  "meta:extensible": false,
+  "meta:abstract": false,
+  "description": "",
+  "definitions": {
+    "attributes": {
+      "properties": {
+        "xdm:color": {
+          "title": "Color",
+          "$ref": "#/definitions/attribute"
+        },
+        "xdm:size": {
+          "title": "Size",
+          "$ref": "#/definitions/attribute"
+        },
+        "xdm:style": {
+          "title": "Style",
+          "$ref": "#/definitions/attribute"
+        },
+        "xdm:weight": {
+          "title": "Weight",
+          "$ref": "#/definitions/attribute"
+        },
+        "xdm:fabrication": {
+          "type": "string",
+          "$ref": "#/definitions/attribute"
+        },
+        "xdm:dimensions": {
+          "title": "Dimensions",
+          "$ref": "#/definitions/attribute"
+        },
+        "xdm:brand": {
+          "title": "Brand name",
+          "$ref": "#/definitions/attribute"
+        },
+        "xdm:manufacturer": {
+          "title": "Manufacturer name",
+          "$ref": "#/definitions/attribute"
+        }
+      },
+      "patternProperties": {
+        ".+://.+": {
+          "$ref": "#/definitions/attribute"
+        }
+      }
+    },
+    "attribute": {
+      "oneOf": [
```

Comment:
@kstreeter - Please look at this.
While I know JSON-Schema allows this, I don't think it is easy for us to represent this form of dynamic data in our applications.  
@mhaack see this: https://github.com/adobe/xdm/blob/master/CONTRIBUTING.md
`avoid unnecessary polymorphism that is hard to consume, e.g. offering both singular and array notations, or introducing type variants that don't provide a common discriminator property.`

Reply:
I guess we can take away the simple option, which will lead to more verbose notation on the wire, but make it a bit easier to consume.

Reply:
I simplified it now by removing the simple attributes as suggested by @trieloff